src/lib/rmt/ssl/certificate_generator.rb

```diff
@@ -58,6 +58,23 @@ def ca_present?
     false
   end
 
+  def ca_encrypted?
+    !valid_password?(' ') # check with emtpy password. password has one char otherwise command requires input
+  end
+
+  def valid_password?(password)
+    RMT::Execute.on_target!(
+      ['echo', password],
```

Comment:
You can directly pass the STDIN content avoiding running extra `echo` subprocess, see https://github.com/openSUSE/cheetah#passing-input

That's also unsafe as somebody could see that in the process list.

There are more places affected below.